Add basic cat command and error handling

pknight24 opened the pull request:
I plan on adding options to the command later. As usual, if there is anything that I need to change, please let me know.

pknight24 commented:
I added some comments that explain how the class works. And if I may ask, what exactly do you mean by writing a test for it? The other command files don't seems to have tests written in. Or do mean in the 'spec' folder?

Sorry if this should be obvious, I am still new to open source contribution.

faraazahmad commented on lib/packages/cat.rb:
```diff
@@ -0,0 +1,18 @@
+#a basic cat command for shellrb
+#TODO: add options/flags
```
Please add a space after # in all comment lines

faraazahmad commented:
Yeah, we write all the ShellRB tests in spec folder. No worries, I'll open the issue for someone to write the tests, Meanwhile I'll add CONTRIBUTING.md for easy understanding.

pknight24 commented:
Thanks for clarifying. I can write the tests in spec, as that's something I should learn about anyway. Adding a CONTRIBUTING.md is a great idea for clearing any confusing for new contributors. If you need any help with that, I would be glad to be of assistance.

pknight24 commented:
I added the spec file and fixed the comment format. I used the RSpec documentation to write cat_spec.rb, so if anything is wrong, please let me know.

faraazahmad commented:
That's cool. The tests don't pass though, unable to find file.txt.
Don't add a file in the repo just for testing, maybe make a file using the code and then delete it after the test completes. Also, with current setup, file.txt cannot be found

faraazahmad commented:
```
$ rake test
rspec
cat: c: No such file or directory
cat: cat_test/file.txt: No such file or directory
Fcat: t: No such file or directory
cat: text.txt: No such file or directory
F.cd: specified path does not exist
.cd: specified path does not exist
.faraaz
faraaz
.
..
.../home
.

Failures:

  1) Cat.main given existing file prints the text in the file
     Failure/Error: expect(Cat.main('cat_test/file.txt')).to eql(Kernel.system 'cat cat_test/file.txt')
     
       expected: false
            got: nil
     
       (compared using eql?)
     # ./spec/packages/cat_spec.rb:9:in `block (4 levels) in <top (required)>'

  2) Cat.main given non-existing file prints error message
     Failure/Error: expect(Cat.main('text.txt')).to eql(Kernel.system 'cat text.txt')
     
       expected: false
            got: nil
     
       (compared using eql?)
     # ./spec/packages/cat_spec.rb:15:in `block (4 levels) in <top (required)>'

Finished in 0.06568 seconds (files took 0.44821 seconds to load)
9 examples, 2 failures

Failed examples:

rspec ./spec/packages/cat_spec.rb:8 # Cat.main given existing file prints the text in the file
rspec ./spec/packages/cat_spec.rb:14 # Cat.main given non-existing file prints error message

rake aborted!
Command failed with status (1): [rspec...]
/home/faraaz/shellrb/Rakefile:13:in `block in <top (required)>'
/home/faraaz/.rvm/gems/ruby-2.4.0/gems/rake-12.1.0/exe/rake:27:in `<top (required)>'
Tasks: TOP => test
(See full trace by running task with --trace)

```

pknight24 commented:
The test works now. Let me know if there are any other changes to make.
